storage/src/main/java/org/apache/kafka/storage/internals/log/AbstractIndex.java

```diff
@@ -420,19 +415,12 @@ protected void truncateToEntries0(int entries) {
         mmap.position(entries * entrySize());
     }
 
-    /**
-     * Execute the given function in a lock only if we are running on windows or z/OS. We do this
-     * because Windows or z/OS won't let us resize a file while it is mmapped. As a result we have to force unmap it
-     * and this requires synchronizing reads.
-     */
-    protected final <T, E extends Exception> T maybeLock(Lock lock, StorageAction<T, E> action) throws E {
-        if (OperatingSystem.IS_WINDOWS || OperatingSystem.IS_ZOS)
-            lock.lock();
+    protected final <T, E extends Exception> T inResizeLock(Lock lock, StorageAction<T, E> action) throws E {
```

Comment:
Perhaps it's better to make this `inResizeReadLock` and avoid passing in lock from the input. For consistency, we could also introduce a method like `inLock` for the other lock. This way, we could make both locks private.